llpc/tool/amdllpc.cpp

```diff
@@ -770,48 +761,52 @@ int main(int argc, char *argv[]) {
 #endif
 #endif
 
-  result = init(argc, argv, &compiler);
-
 #ifdef WIN_OS
   if (AssertToMsgBox) {
     _set_error_mode(_OUT_TO_MSGBOX);
   }
 #endif
 
-  // Simplify error handling and enable early returns. These assume that result statuses
-  // are always written to the |result| local variable.
+  Result result = Result::Success;
+  ICompiler *compiler = nullptr;
+  result = init(argc, argv, compiler);
+
+  // Simplify error handling and enable early returns. Assume that result statuses are always written to the `result`
+  // local variable.
   auto isFailure = [&result] { return result != Result::Success; };
```

Comment:
This is pre-existing, but while you're cleaning this up: this lambda is really not a good abstraction in my opinion. It would be better to just put `result != Result::Success` in the relevant places. That's not particularly long, is more explicit, and doesn't use magically bound state as much.